exist-core/src/main/java/org/exist/xquery/functions/fn/FunHigherOrderFun.java

```diff
@@ -102,13 +104,14 @@ public FunHigherOrderFun(XQueryContext context, FunctionSignature signature) {
 	}
 
     @Override
-    protected void checkArguments() throws XPathException {
+	protected void checkArgument(final Expression arg, @Nullable final SequenceType argType,
+			final AnalyzeContextInfo argContextInfo, final int argPosition) throws XPathException {
         // hack: order of parameters for filter and other functions has changed
```

Comment:
Is this comment still valid?

Reply:
No idea. I didn't add the comment or implement the filter stuff - you would have to ask the author of that stuff.

Reply:
But it _clearly_ does not comment what you added and that is the entire method body. Please remove it as it will only confuse future generations.